compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2228,17 +2240,18 @@ private InferredTupleDetails checkExprList(List<BLangExpression> exprs, BType ex
     }
 
     private static class InferredTupleDetails {
-        List<BType> fixedMemberTypes = new ArrayList<>();
+        List<BTupleMember> fixedMemberTypes = new ArrayList<>();
```

Comment:
Is this change required? Isn't it enough to have the type?

Reply:
Not cast, right? Create new symbols? Since we are creating a new tuple type with this information, IMO, we should have just the types and create new symbols/tuple members? For example, if `org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker#updateInferredTupleDetailsFromSpreadMember` is called, and members from `spreadOpExprType` are added to `fixedMemberTypes` it's wrong to directly use those members, right? We need to create new members for the new tuple type?

Reply:
Updated to create new symbols.